### DISCUSSION TITLE
Use gptel programmatically

### DISCUSSION THREAD

**Discussion by pprevos** [karthink/gptel#532] [Q&A]
Hi, I like to use gptel programmatically in my own package. Something like: 

```
(setq title (gptel-prompt "Generate a title for a book about ..."))
```

I don't want any output inserted in the active buffer or any of the standard prompts in gptel being used, just the API mechanisms.

P:)

**Comment by karthink** [comment]
Yes, you can use `gptel-request` and supply a callback. See the function documentation for details and the [wiki](https://github.com/karthink/gptel/wiki/Defining-custom-gptel-commands) for examples.

However `gptel-request` is asynchronous so you'll have to structure it differently.

``` emacs-lisp
(defvar title nil)

(gptel-request "Generate a title for a book about..."
  :callback (lambda (response _)
              (when (stringp response)
                (setq title response))))
```

**Reply by pprevos** [reply]
Thanks, makes more sense now.
